code/backend/batch/BatchPushResults.py

```diff
@@ -27,21 +27,38 @@ def _get_file_name_from_message(msg: func.QueueMessage) -> str:
     arg_name="msg", queue_name="doc-processing", connection="AzureWebJobsStorage"
 )
 def batch_push_results(msg: func.QueueMessage) -> None:
-    do_batch_push_results(msg)
+    message_body = json.loads(msg.get_body().decode("utf-8"))
+    logger.info("Process Document Event queue function triggered: %s", message_body)
+
+    event_type = message_body.get("eventType", "")
+    # We handle "" in this scenario for backwards compatibility
+    # This function is primarily triggered by an Event Grid queue message from the blob storage
+    # However, it can also be triggered using a legacy schema from BatchStartProcessing
+    if event_type in ("", "Microsoft.Storage.BlobCreated"):
+        _process_document_created_event(message_body)
 
+    elif event_type == "Microsoft.Storage.BlobDeleted":
+        _process_document_deleted_event(message_body)
 
-def do_batch_push_results(msg: func.QueueMessage) -> None:
+    else:
+        logger.error("Unknown event type received: %s", event_type)
+        raise NotImplementedError(f"Unknown event type received: {event_type}")
```

Comment:
suggestion: generally, logging an error and throwing an exception is considered an anti-pattern.
Since, throwing an exception will inform the user of this function that something has gone wrong, logging an error doesn't provide an additional value, instead logger.info can be used to log meaningful information for the debugging activity. Also, if there are log-scanners (generally, there should be) to raise alarms/tickets on error log count, then we get notified of 1 error 2 times - once through the thrown exception and second from log scanning.

Reply:
good spot - thanks, will update